internal/message/types_test.go

```diff
@@ -0,0 +1,122 @@
+package message
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+func TestProviderResources(t *testing.T) {
+	resources := new(ProviderResources)
+	gc1 := &gwapiv1b1.GatewayClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-gc1",
+		},
+	}
+	gw1 := &gwapiv1b1.Gateway{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test1",
+			Namespace: "test",
+		},
+	}
+	r1 := &gwapiv1b1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "route1",
+			Namespace: "test",
+		},
+	}
```

Comment:
Can you add a test case where gatewayclasses, gateways, and httproutes are nil?

xref: https://github.com/envoyproxy/gateway/pull/216